### DISCUSSION TITLE
注册多个消息处理器，怎么停止冒泡

### DISCUSSION THREAD

**Discussion by sonhineboy** [w7corp/easywechat#2309] [Q&A]
代码如果下
```
$app->server->push(MessageLogHandler::class); // h1
$app->server->push(MessageReplyHandler::class); // h2
$app->server->push(OtherHandler::class);// h3
$app->server->push(...);// h3
```
目前看来，这个事观察者，也就是说只要push上去，都会执行（h1-hn）;有没有那种，就像js事件那种，禁止向上冒泡那种操作，如果 h1 里面有禁止冒泡的操作，h2 、h3、hn都不再执行

**Comment by overtrue** [comment]
1. 加条件 where()
2. 返回非空 & FinalResult  https://github.com/w7corp/easywechat/blob/5.x/src/Kernel/Traits/Observable.php#L159-L162
